design/pipeline-resources.md

```diff
@@ -83,7 +83,7 @@ Or to avoid downloading any of the artifacts at all:
 - download: none
 ```
 
-Artifacts from the `pipeline` resource are downloaded to `$PIPELINES_RESOURCESDIR/<pipeline-identifier>/<artifact-identifier>` folder; see [artifact download location](https://github.com/Microsoft/azure-pipelines-yaml/blob/master/design/pipeline-artifacts.md#artifact-download-location) for more details.
+Artifacts from the `pipeline` resource are downloaded to `$(PIPELINE.WORKSPACE)/<pipeline-identifier>/<artifact-identifier>` folder; see [artifact download location](https://github.com/Microsoft/azure-pipelines-yaml/blob/master/design/pipeline-artifacts.md#artifact-download-location) for more details.
```

Comment:
what if there are folder name conflicts? for example, the current pipeline has an artifact named `foo`, and there is a `foo` pipeline resource ...

Reply:
The name of the artifact published by the current pipeline is set in the YAML. Pipeline identifier is a unique alias for the pipeline resource user provides in the YAML. User needs to ensure the pipeline identifier is unique to avoid collisions. These are very corner scenarios and can be avoided by the user during the pipeline YAML authoring time. In future we can provide queue time validations and warn users of possibility for collisions. 
We want to provide a simpler experience for CICD and CD scenarios.